File: src/main/java/ru/itis/fisd_cw/data/entity/NoteEntity.java
package ru.itis.fisd_cw.data.entity;


import jakarta.annotation.PreDestroy;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Builder
@Entity
@Table(name = "notes")
@Data
@Getter
@Setter
@RequiredArgsConstructor
@AllArgsConstructor
public class NoteEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "title", nullable = false, length = 32)
    String title;


    @Column(name = "content", nullable = false, length = 128)
    String content;



    //Нужно было snakecase, но я забыл и в бд сделал кемел
    @Column(name = "createdat", nullable = false)
    LocalDateTime creationAt;


    //Нужно было snakecase, но я забыл и в бд сделал кемел
    @Column(name = "updatedat")
    LocalDateTime updatedAt;


    //Нужно было snakecase, но я забыл и в бд сделал кемел
    @Column(name = "deletedat")
    LocalDateTime deletedAt;

    @PreUpdate
    protected void preUpdate() {
        setUpdatedAt(LocalDateTime.now());
    }

    public void delete() {
        setDeletedAt(LocalDateTime.now());
    }

    @PrePersist
    protected void onCreate() {
        setCreationAt(LocalDateTime.now());
    }

}

File: src/main/kotlin/com/example/demo/entities/NoteEntity.kt
package com.example.demo.entities

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "notes")
class NoteEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,

    @Column(nullable = false, length = 255)
    var title: String = "",

    @Column(nullable = false, columnDefinition = "TEXT")
    var content: String = "",

    @Column(nullable = false, name = "created_at")
    var createdAt: Long = System.currentTimeMillis(),

    @Column(nullable = false, name = "updated_at")
    var updatedAt: Long = System.currentTimeMillis(),

    @Column(name = "deleted_at")
    var deletedAt: Long? = null,
)
